Skip over unique constraint duplicate record errors, adjust error logging in index_setsm.py

ehusby opened the pull request:
-- IMPORTANT NOTE --
The most important change in this PR is in how index_setsm.py handles the "duplicate key value violates unique constraint" error when add records to the target table. Previously, the entire index_setsm.py process would error out upon hitting the first record that trips the unique constraint error. Now it will just print error messages and skip over those records.


See individual commits for details on the particular tweaks.

This needs more testing, but I think these changes touch on some really important areas for improvement in the error handling.

I believe NASA has been running with these changes applied as a patch to their pgcdemtools code.

clairecporter commented:
I merged the master into this to test.  I'll report back here shortly on the result of the testing.

clairecporter commented:
Tests passed.  You are good to merge.

ehusby commented:
@clairecporter I changed the PR name to more accurately record the major change here, and added this note to the PR description. Just FYI:
```
-- IMPORTANT NOTE --
The most important change in this PR is in how index_setsm.py handles the "duplicate key value violates unique constraint" error when add records to the target table. Previously, the entire index_setsm.py process would error out upon hitting the first record that trips the unique constraint error. Now it will just print error messages and skip over those records.
```

Still good to merge this in?
